.github/actions/run-tests/action.yml

```diff
@@ -0,0 +1,34 @@
+name: Run tests and report results
+inputs:
+  check-pyarrow-version:
+    required: false
+runs:
+  using: composite
+  steps:
+    - name: Check PyArrow version
+      run: |
+        # Double check that we have the expected PyArrow
+        pushd /tmp
+        python -c "import pandas; pandas.show_versions()" | egrep -i "pyarrow.+: ${{ matrix.check-pyarrow-version }}"
```

Comment:
Maybe better to check from conda instead of show_version as show_versions may be more prone to break?

Reply:
Fine with me, this is taken from other GHA recipes

https://github.com/pandas-dev/pandas/blob/79fb2debb14e77d6d4af9c4db058e6f994507e29/.github/workflows/sdist.yml#L91

https://github.com/pandas-dev/pandas/blob/79fb2debb14e77d6d4af9c4db058e6f994507e29/.github/workflows/posix.yml#L177

https://github.com/pandas-dev/pandas/blob/main/.github/workflows/python-dev.yml#L74